scipy/integrate/tests/test_quadrature.py

```diff
@@ -659,9 +659,45 @@ def test_special_cases(self):
 
         # Should add tests of:
         # - all elements of `x` identical
-        # - y of size 1 and 2
         # These should work as they do for `simpson`
 
+    @pytest.mark.parametrize('axis', np.arange(-3, 3))
+    @pytest.mark.parametrize('x_ndim', (1, 3))
+    @pytest.mark.parametrize('x_len', (1, 2))
+    @pytest.mark.parametrize('i_ndim', (None, 0, 3))
+    @pytest.mark.parametrize('dx', (None, True))
+    def test_y_of_size_under_3(self, axis, x_ndim, x_len, i_ndim, dx):
```

Comment:
Rather than adding a whole other test, can you just add the desired behavior to `cumulative_simpson_reference` and use the existing test?